app/controllers/orders_controller.rb

```diff
@@ -17,7 +17,7 @@ def create
   def edit
     @order = Order.find(params[:id])
 
-    if @order.order_submitted? && !current_user.super_admin?
+    if load_edit?
       redirect_to orders_path
     elsif Rails.root.join("app/views/orders/status/#{@order.status}.html.erb").exist?
       render "orders/status/#{@order.status}"
```

Comment:
Does this (or should it) have some kind of `current_user.can_view_order?(@order)` logic?